src/corpus/ui/ChapterViewLine.tsx

```diff
@@ -204,43 +240,46 @@ export function ChapterViewLineVariant({
     variant,
   ])
 
+  const [, highlightIndexSetter] = useState(0)
+  const lineGroup = useMemo(() => {
```

Comment:
Wee man manuell rerender triggered bei React ohne es zu hacken

Reply:
Eine Klasse scheint nicht der präferierte Weg zu sein, komplexe Informationen in React zwischen verschiedenen Components zu teilen. Man müsste scheinbar eigentlich einen custom Hook machen und die ganze `LineGroup`-Logik dort unterbringen, mit einem State für den `highlightTokenIndex` (vgl. z.B. [hier](https://stackoverflow.com/a/66063150/15873043)). Ein "leerer" state setter ist sonst scheinbar mit die einzige Möglichkeit, einen re-render in Function components zu forcen (s. [hier](https://stackoverflow.com/a/46240804/15873043)).

Reply:
das einfach als kommentar hilfreich für später mit links für useReduce zum beispiel.